nix/devcontainer/devcontainer.nix

```diff
@@ -74,27 +83,31 @@ let
       # make sure /tmp exists
       mkdir -m 1777 tmp
 
-      # need a HOME
-      mkdir -vp root
-
       # allow ubuntu ELF binaries to run. VSCode copies it's own.
       chmod +w lib64
       ln -s ${glibc}/lib64/ld-linux-x86-64.so.2 lib64/ld-linux-x86-64.so.2
       ln -s ${gcc-unwrapped.lib}/lib64/libstdc++.so.6 lib64/libstdc++.so.6
       chmod -w lib64
-
     '' + extraCommands;
 
+    runAsRoot = ''
+      #!${runtimeShell}
```

Comment:
Looking at `dockerTools`, they already pass this through something that adds the shebang, so you can get away without it.